File: 7_kyu/Get_the_Middle_Character.jl
"""
  Returns the one or two characters in the string middle.
"""
function getmiddle(s::String)::String
  if length(s) < 2
    # Special case: empty string.
    # Also: 1 byte strings fast exit.
    return s
  end
  # Converts a string of bytes, UTF-8 encoded, into an array of characters.
  chars = collect(s)
  len = length(chars)
  String(chars[(len+1)÷2:1+len÷2])
end

File: 7_kyu/Get_the_Middle_Character.r
# Returns the one or two characters in the middle of string.
get_middle <- function(s){
  # Number of characters in the string, UTF-8 save.
  len <- stringr::str_length(s)
  if( len < 3 ) return(s)
  stringr::str_sub(s, (len+1) %/% 2, (len+2) %/% 2 )
}
